OneCommunityGlobal/HighestGoodNetworkApp
# Tim: BM Dashboard - Add Type: Equipment

# Description
Implements Phase 2 WBS # 9.2.7: Adds new view **Add Type: Equipment** with form

## Related PRS (if any):
Checkout [backend PR 677](https://github.com/OneCommunityGlobal/HGNRest/pull/677) to test this branch

Please add a review for that PR as well!

## Main changes explained:
- Adds view **Add Type: Equipment** with form
- Form initiates a POST request to add a "equipment" type to the database

## How to test:
1. `git pull` development branch to make sure you are up to date
2. `git checkout Tim_add_equip_form`
3. `nvm i`, then `npm i`
4. `npm run start:local`
6. Clear site data/cache, or use a fresh user profile in your browser
7. Log in as volunteer user
8. Navigate to http://localhost:3000/bmdashboard/login to confirm your login credentials
9. Navigate to http://localhost:3000/bmdashboard/equipment/add
10. Use the form to create a new "equipment" type item. For now we're not defining what counts as equipment, so any entry is acceptable. Test that the Submit button is disabled if any inputs are blank. Test max character limit in the description input.
11. After creating a new item type, try to create another item with precisely the same name to receive a 409 error from the server.
12. Test that the Cancel button returns user to previous page.

## Demo:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/be5c6733-1fa1-478a-b3c2-01a1edff2284

## Note:
This branch also adds the view "/bmdashboard/equipment", which is currently empty. There is also some code for implementing a Check Existing Types" modal that will be finished in a separate branch.
